src/cryptonotecore/Core.cpp

```diff
@@ -0,0 +1,3549 @@
+
+        const auto [success, error] = isTransactionValidForPool(cachedTransaction, validatorState);
+        if (!success)
+        {
+            return {false, error};
+        }
+
+        auto transactionHash = cachedTransaction.getTransactionHash();
+
+        if (!transactionPool->pushTransaction(std::move(cachedTransaction), std::move(validatorState)))
+        {
+            logger(Logging::DEBUGGING) << "Failed to push transaction " << transactionHash
+                                       << " to pool, already exists";
+            return {false, "Transaction already exists in pool"};
+        }
+
+        logger(Logging::DEBUGGING) << "Transaction " << transactionHash << " has been added to pool";
+        return {true, ""};
+    }
+
+    std::tuple<bool, std::string> Core::isTransactionValidForPool(
+        const CachedTransaction &cachedTransaction,
+        TransactionValidatorState &validatorState)
+    {
+        auto [success, err] = Mixins::validate({cachedTransaction}, getTopBlockIndex());
+
+        if (!success)
+        {
+            return {false, "Transaction does not contain the proper number of ring signatures"};
+        }
+
+        if (cachedTransaction.getTransaction().extra.size() >= CryptoNote::parameters::MAX_EXTRA_SIZE_V3)
+        {
+            logger(Logging::TRACE) << "Not adding transaction " << cachedTransaction.getTransactionHash()
+                                   << " to pool, extra too large.";
+
+            return {false, "Transaction extra data is too large"};
+        }
+
+        uint64_t fee;
+
+        if (auto validationResult =
+                validateTransaction(cachedTransaction, validatorState, chainsLeaves[0], fee, getTopBlockIndex()))
+        {
+            logger(Logging::DEBUGGING) << "Transaction " << cachedTransaction.getTransactionHash()
+                                       << " is not valid. Reason: " << validationResult.message();
+            return {false, validationResult.message()};
+        }
+
+        auto maxTransactionSize = getMaximumTransactionAllowedSize(blockMedianSize, currency);
+        if (cachedTransaction.getTransactionBinaryArray().size() > maxTransactionSize)
+        {
+            logger(Logging::WARNING) << "Transaction " << cachedTransaction.getTransactionHash()
+                                     << " is not valid. Reason: transaction is too big ("
+                                     << cachedTransaction.getTransactionBinaryArray().size()
+                                     << "). Maximum allowed size is " << maxTransactionSize;
+            return {false, "Transaction size (bytes) is too large"};
+        }
+
+        bool isFusion = fee == 0
+                        && currency.isFusionTransaction(
+                            cachedTransaction.getTransaction(),
+                            cachedTransaction.getTransactionBinaryArray().size(),
+                            getTopBlockIndex());
+
+        const uint64_t minFee = 
```

Comment:
This is not exactly a bug per-se, but it does mean that full block validation is not performed. The reason there is no `if (blockIndex >= CryptoNote::parameters::MAX_EXTRA_SIZE_V1_HEIGHT` is because previously the extra size was unlimited, and so we just had the `V2` check here.

If we're introducing a `V3` we need to keep the `V2` check otherwise it could potentially allow a [very very old] alt chain to be considered valid when it has a tx extra size > 1024 bytes.

The correct code should be something like

```cpp
if (blockIndex >= CryptoNote::parameters::MAX_EXTRA_SIZE_V3_HEIGHT + CryptoNote::parameters::CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW)
{
    if (transaction.extra.size() >= CryptoNote::parameters::MAX_EXTRA_SIZE_V3)
    {
        return error::TransactionValidationError::EXTRA_TOO_LARGE;
    }
}
else if (blockIndex >= CryptoNote::parameters::MAX_EXTRA_SIZE_V2_HEIGHT + CryptoNote::parameters::CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW)
{
    if (transaction.extra.size() >= CryptoNote::parameters::MAX_EXTRA_SIZE_V2)
    {
        return error::TransactionValidationError::EXTRA_TOO_LARGE;
    }
}
```

Or if you wanted to be fancy..

```cpp
uint64_t maxExtraSize = std::numeric_limits<uint64_t>::max();

if (blockIndex >= CryptoNote::parameters::MAX_EXTRA_SIZE_V3_HEIGHT + CryptoNote::parameters::CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW)
{
    maxExtraSize = CryptoNote::parameters::MAX_EXTRA_SIZE_V3;
}
else if (blockIndex >= CryptoNote::parameters::MAX_EXTRA_SIZE_V2_HEIGHT + CryptoNote::parameters::CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW)
{
    maxExtraSize = CryptoNote::parameters::MAX_EXTRA_SIZE_V2;
}

if (transaction.extra.size() >= maxExtraSize)
{
    return error::TransactionValidationError::EXTRA_TOO_LARGE;
}
```

Again, this isn't really going to effect anything, but should probably be fixed anyway.